src/Microsoft.AspNetCore.Http.Abstractions/Authentication/AuthenticateInfo.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Claims;
+
+namespace Microsoft.AspNetCore.Http.Authentication
+{
+    /// <summary>
+    /// Used to store the results of an Authenticate call.
+    /// </summary>
+    public class AuthenticateInfo
+    {
+        /// <summary>
+        /// The <see cref="ClaimsPrincipal"/>.
+        /// </summary>
+        public ClaimsPrincipal Principal { get; set; }
```

Comment:
Curious: is there any particular reason to use publicly settable properties instead of get-only properties + public constructor?